### PR TITLE
wesnoth: 1.14.11 -> 1.14.12

### PR DESCRIPTION
**Pull request by kampka**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kampka** [comment]
@GrahamcOfBorg build wesnoth

**Review by puzzlewolf — CHANGES_REQUESTED** [review]
:+1: for moving to the github source. I just found one thing :)

This PR replaces #88777 (automatic update).

File: pkgs/games/wesnoth/default.nix [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
-  version = "1.14.11";
+  version = "1.14.12";
 
-  src = fetchurl {
-    url = "mirror://sourceforge/sourceforge/${pname}/${pname}-${version}.tar.bz2";
-    sha256 = "1i8mz6gw3qar09bscczhki0g4scj8pl58v85rp0g55r4bcq41l5v";
+  src = fetchFromGitHub {
+    rev = version;
+    owner = pname;
+    repo = pname;
```
[the remark this answers is not in the document]
Comment (kampka) [reply pkgs/games/wesnoth/default.nix]: agreed :+1:

**Comment by puzzlewolf** [comment]
Can you move the `meta.homepage` from `http` to `https`, too?

**Comment by kampka** [comment]
> Can you move the `meta.homepage` from `http` to `https`, too?

Done

**Review by puzzlewolf — APPROVED** [review]
LGTM :+1:

Both binaries start, played a bit of the tutorial :)

**Pull request merged**